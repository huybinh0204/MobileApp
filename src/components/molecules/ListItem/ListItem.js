import PropTypes from 'prop-types';
import React from 'react';
import { IconSvg } from '_components/atoms';
import { COLORS } from '_constants';
import { normalize } from '_utilities/screen';
import { Container, Icon, LeftContainer, SubTitle, TextContainer, Title } from './ListItem.styles';

const ListItem = ({ children, color, icon, title, subtitle }) => {
  return (
    <Container>
      <LeftContainer>
        {icon && (
          <Icon testID="LeftIcon">
            <IconSvg width={normalize(30)} height={normalize(30)} icon={icon} />
          </Icon>
        )}
        <TextContainer>
          <Title color={color} numberOfLines={1}>
            {title}
          </Title>
          {subtitle && (
            <SubTitle color={color} numberOfLines={2}>
              {subtitle}
            </SubTitle>
          )}
        </TextContainer>
      </LeftContainer>
      {children}
    </Container>
  );
};

ListItem.defaultProps = {
  children: null,
  color: COLORS.alpha500,
  icon: null,
  subtitle: null,
};

ListItem.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
  icon: PropTypes.string,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

export default ListItem;
