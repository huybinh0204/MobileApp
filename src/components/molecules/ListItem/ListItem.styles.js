import styled from '@emotion/native';
import { BaseText } from '_components/atoms';

export const Container = styled.View({
  alignSelf: 'stretch',
  flexDirection: 'row',
  alignItems: 'center',
  justifyContent: 'space-between',
});

export const LeftContainer = styled.View({
  flex: 1,
  flexDirection: 'row',
  alignItems: 'center',
});

export const Icon = styled.View(({ theme }) => ({
  marginRight: theme.spacing.s,
}));

export const Title = styled(BaseText)(({ theme, color }) => ({
  ...theme.typography.subContentSemiBold,
  color: color,
  width: '80%',
}));

export const SubTitle = styled(BaseText)(({ theme, color }) => ({
  ...theme.typography.tinyContentRegular,
  color: color,
  width: '80%',
}));

export const TextContainer = styled.View(() => ({
  width: '100%',
}));
