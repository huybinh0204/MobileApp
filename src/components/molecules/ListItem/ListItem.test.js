import { render } from '@testing-library/react-native';
import React from 'react';
import { ICONS } from '_constants';
import Theme from '_utilities/Theme';
import ListItem from './ListItem';

describe('ListItem', () => {
  let props;

  const RederListItem = (innerProps) => (
    <Theme>
      <ListItem {...innerProps} />
    </Theme>
  );

  beforeEach(() => {
    props = {
      title: 'List Item',
      icon: ICONS.logo,
    };
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(<RederListItem {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should display the title', () => {
    const { getByText } = render(<RederListItem {...props} />);
    expect(getByText(props.title)).not.toBeNull();
  });

  it('should display the icon', () => {
    const { queryByTestId } = render(<RederListItem {...props} />);
    expect(queryByTestId('LeftIcon')).not.toBeNull();
  });

  it('should not display the icon', () => {
    const { queryByTestId } = render(<RederListItem title={props.title} />);
    expect(queryByTestId('LeftIcon')).toBeNull();
  });
});
