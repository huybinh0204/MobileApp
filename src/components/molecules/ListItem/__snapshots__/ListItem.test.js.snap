// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 29,
          },
          undefined,
        ]
      }
      testID="LeftIcon"
    >
      <SvgMock
        height={54.5}
        width={54.5}
      />
    </View>
    <View
      style={
        Array [
          Object {
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        color="#3D3397"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#3D3397",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 25.5,
              "lineHeight": 35.5,
              "width": "80%",
            },
            undefined,
          ]
        }
      >
        List Item
      </Text>
    </View>
  </View>
</View>
`;
